hooks/open-telemetry/pkg/metrics.go

```diff
@@ -0,0 +1,100 @@
+package otel
```

Comment:
Hmmm... what you did here was _not_ introduce another package, but add the hook to the existing package... I never considered that solution honestly, in my mind, we'd add a totally new package that would be a different dependency.

I think there's merit to this approach though, in this repo as well as others. We could put the metrics and traces hooks in the same artifact (hooks.otel) and just provide 2 different hooks in that jar/module, etc.

Perhaps the only downside would be that the metrics AND the traces OTel deps would be bundled - but I don't even know if that's much of an issue since they would be tree shaken or not compiled in many cases.

Reply:
@beeme1mr see above.

Reply:
Seems reasonable to me. At least in Go.

Reply:
@Kavindu-Dodan I think we may want to rename `otel.go` and `NewHook` to `traces.go` and `NewTraceHook` respectively. Yes it will be a breaking change, but we are <0 and it's a very easy change to absorb.

I dont mind whether it's done in this PR or another, but IMO we should do this before we release this new stuff.

Reply:
Yeah agree. This can be a follow-up PR, but should prior to the release as you said :)

Reply:
Created a follow-up issue https://github.com/open-feature/go-sdk-contrib/issues/224